File: content-solutions/greedy/biggest-after-erase.py
def solution(number, k):
    stack = []
    for d in number:
        while k > 0 and stack and stack[-1] < d:
            stack.pop()
            k -= 1
        stack.append(d)
    if k:
        stack = stack[:-k]
    return "".join(stack)

File: content-solutions/greedy/biggest-after-erase.js
function solution(number, k) {
  const stack = [];
  for (const d of number) {
    while (k > 0 && stack.length && stack[stack.length - 1] < d) {
      stack.pop();
      k--;
    }
    stack.push(d);
  }
  stack.length -= k;
  return stack.join("");
}
